roles/openshift_aws/tasks/build_ami.yml

```diff
@@ -0,0 +1,48 @@
+---
+- when: openshift_aws_create_vpc | bool
+  name: create a vpc
+  include: vpc.yml
+
+- when: openshift_aws_users | length  > 0
+  name: create aws ssh keypair
+  include: ssh_keys.yml
+
+- when: openshift_aws_create_security_groups | bool
+  name: Create compute security_groups
+  include: security_group.yml
+
+- name: query vpc
+  ec2_vpc_net_facts:
+    region: "{{ openshift_aws_region }}"
+    filters:
+      'tag:Name': "{{ openshift_aws_vpc_name }}"
+  register: vpcout
+
+- name: fetch the default subnet id
+  ec2_vpc_subnet_facts:
+    region: "{{ openshift_aws_region }}"
+    filters:
+      "tag:Name": "{{ openshift_aws_subnet_name }}"
+      vpc-id: "{{ vpcout.vpcs[0].id }}"
+  register: subnetout
+
+- name: create instance for ami creation
+  ec2:
+    assign_public_ip: yes
+    region: "{{ openshift_aws_region }}"
+    key_name: "{{ openshift_aws_ssh_key_name }}"
+    group: "{{ openshift_aws_clusterid }}"
+    instance_type: m4.xlarge
+    vpc_subnet_id: "{{ subnetout.subnets[0].id }}"
```

Comment:
This should be configurable if I want to specify the subnet ID myself.

Reply:
The way this works is that you can specify the subnet name and we query it.  Then this chooses the first subnet id from the name that was passed in `openshift_aws_subnet_name`.

We can make the subnet id a variable and then skip the query if passed.

Reply:
@michaelgugino, I'm fine with going back through and adding more options for passing variables but this PR is getting too large and I'd like to get the functionality merged since its just a refactor of the roles that are currently in the repo.

Reply:
You're right, this one works as described.  I was hitting an error but I figured it out.  Other comments I think are necessary.